### ISSUE TITLE
AccessTokenPlugin not working in Unit Test

### ISSUE THREAD

**Issue by hallelujahbaby** [Moya/Moya#1918]
Moya Version: 14.0.0-beta.2

```swift
import Quick
import Nimble
import Moya

class Testing: QuickSpec {

    struct TestTarget: TargetType, AccessTokenAuthorizable {
        let baseURL = URL(string: "http://www.api.com/")!
        let path = ""
        let method = Method.get
        let task = Task.requestPlain
        let sampleData = Data()
        let headers: [String: String]? = nil
        let authorizationType: AuthorizationType
    }

    override func spec() {

        var sut: MoyaProvider<TestTarget>!

        beforeEach {
            let token = UUID().uuidString
            let authPlugin = AccessTokenPlugin { token }
            sut = MoyaProvider<TestTarget>(stubClosure: MoyaProvider.immediatelyStub,
                                           plugins: [authPlugin])
        }

        it("test") {
            var header: String?
            _ = sut.rx.request(TestTarget(authorizationType: .bearer))
                      .subscribe { event in
                          switch event {
                              case .success(let response):
                                  header = response.request?.value(forHTTPHeaderField: "Authorization")
                              case .error:
                                  fail()
                          }
                      }

            expect(header).toNot(beNil())
        }
    }
}
```
Testing Result `expected to not be nil, got <nil>`.

I got confused that the plugin seems not working in Stub mode, but when I connect to real network the plugin work as usual.

My question: Is it normal that plugin is not working in test ? Or am I testing in wrong way ?
Thanks.

**Comment by sunshinejr** [comment]
Hey @hallelujahbaby. Did it start breaking with Moya 14 (meaning it was working correctly on 13)?

Also, try replacing `toNot` with `toEventuallyNot` as the assigning should happen asynchronously.

**Comment by hallelujahbaby** [comment]
Hey @sunshinejr , thx for your reply.

I don't know if it is working on 13, cuz this is my first time to write test on Moya 😐, I may try if i have spare time.

I tried to replace `toNot` with `toEventuallyNot` but still no luck.

**Comment by hallelujahbaby** [comment]
I just tested with Moya 13, the test passed.
I can confirm that it started breaking with Moya 14.

**Comment by sunshinejr** [comment]
okay, thanks a lot for the report @hallelujahbaby, gonna investigate and comment here once I have something to share!

**Comment by sunshinejr** [comment]
Hey @hallelujahbaby, just FYI I reproduced the issue, found the problem and I'm currently working on the fix.

**Comment by hallelujahbaby** [comment]
Cool, Thanks. 😄

**Comment by sunshinejr** [comment]
Hey @hallelujahbaby - I just released 14.0.0-beta.3 with the fix. Please try this one and reopen if you still have problems. Thanks again for bringing it up!
